File: windows/ReviewAnything/Models/ReviewItem.cs
namespace ReviewAnything.Models;

public class ReviewItem
{
    public int Id { get; set; }
    public int ChunkId { get; set; }
    public Chunk Chunk { get; set; } = null!;
    public string Question { get; set; } = "";
    public string Answer { get; set; } = "";
    public bool IsHard { get; set; }
    public int ReviewCount { get; set; }
    public DateTime NextReviewAt { get; set; } = DateTime.UtcNow;
    public DateTime? LastReviewedAt { get; set; }
    public bool LlmFailed { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: android/app/src/main/java/com/reviewanything/app/data/model/ReviewItem.kt
package com.reviewanything.app.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "review_items",
    foreignKeys = [
        ForeignKey(
            entity = Chunk::class,
            parentColumns = ["id"],
            childColumns = ["chunkId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["chunkId"])]
)
data class ReviewItem(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val chunkId: Int = 0,
    val question: String = "",
    val answer: String = "",
    val isHard: Boolean = false,
    val reviewCount: Int = 0,
    val nextReviewAt: Long = System.currentTimeMillis(),
    val lastReviewedAt: Long? = null,
    val llmFailed: Boolean = false,
    val createdAt: Long = System.currentTimeMillis()
)
